app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt

```diff
@@ -93,18 +93,36 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
 
         dialogBuilder.setTitle(R.string.location_access)
         dialogBuilder.setMessage(R.string.location_access_message)
-        dialogBuilder.setPositiveButton("NEXT", { dialog, whichButton ->
+        dialogBuilder.setPositiveButton(R.string.next, { dialog, whichButton ->
 
             val callGPSSettingIntent = Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS)
             startActivityForResult(callGPSSettingIntent, 0)
 
         })
-        dialogBuilder.setNegativeButton("Cancel", { dialog, whichButton ->
+        dialogBuilder.setNegativeButton(R.string.cancel, { dialog, whichButton ->
         })
         val b = dialogBuilder.create()
         b.show()
     }
 
+    override fun showWifiIntentDialog() {
+        var dialog: AlertDialog?
+        var dialogBuilder = AlertDialog.Builder(activity as DeviceActivity)
+        dialogBuilder.setView(R.layout.select_dialog_item_material)
+
+        dialogBuilder.setTitle(R.string.wifi_access);
+        dialogBuilder.setMessage(R.string.wifi_access_message)
+        dialogBuilder.setPositiveButton(R.string.next) { dialog, whichButton ->
+            val wifiIntent = Intent(Settings.ACTION_WIFI_SETTINGS)
+            startActivityForResult(wifiIntent, 1)
+        }
+        dialogBuilder.setNegativeButton(R.string.cancel) { dialog, whichButton ->
+            dialog.dismiss()
+        }
+        dialog = dialogBuilder.create()
+        dialog.show()
```

Comment:
Remove last 2 lines and directly call show

Reply:
@iamareebjamal I did this because you said to dismiss the dialog in the press of cancel.

Reply:
You are receiving dialog in the callback